store/models.py

```diff
@@ -587,13 +562,24 @@ class OptionProduct(BaseProduct):
         default=TYPE_COMMON,
     )
 
+    @property
+    def remaining_quantity(self):
+        remaining_quantity = self.stock
+        if self.manage_stock:
+            from retirement.models import Reservation
+            refunded_order_lines = Reservation.objects.filter(is_active=False).values_list('order_line', flat=True)
+            remaining_quantity = OrderLineBaseProduct.objects.filter(option=self)\
+                .exclude(order_line__in=refunded_order_lines)\
+                .aggregate(sum=Sum('quantity'))['sum']
+        return remaining_quantity if remaining_quantity else self.stock
+
```

Comment:
`remaining_quantity` is not available by default on the OptionProduct serializer so frontend will not get the value.
You need to add it in the serializer and make sure that when frontend list Retreat they get this value on the list of Option of the retreats details.

Reply:
I updated the serializer